src/hooks/useOnEscapePress.ts

```diff
@@ -1,22 +1,60 @@
 import {useEffect, useCallback} from 'react'
 
-export type UseOnEscapePressSettings = {
-  onEscape: (e: KeyboardEvent) => void
-}
+const handlers: ((e: KeyboardEvent) => void)[] = []
 
-export const useOnEscapePress = ({onEscape}: UseOnEscapePressSettings): void => {
-  const handleEscape = useCallback(
-    (e: KeyboardEvent) => {
-      if (e.key === 'Escape' && !e.defaultPrevented) {
-        onEscape(e)
+/**
+ * Sets up a `keydown` listener on `window.document`. If
+ * 1) The pressed key is "Escape", and
+ * 2) The event has not had `.preventDefault()` called
+ * The given callback will be executed.
+ *
+ * Note: If multiple `useOnEscapePress` hooks are active simultaneously, the
+ * callbacks will occur in reverse order. In other words, if a parent component
+ * and a child component both call `useOnEscapePress`, when the user presses
+ * Escape, the child component's callback will execute, followed by the parent's
+ * callback. Each callback has the chance to call `.preventDefault()` on the
+ * event to prevent further callbacks.
+ *
+ * @param callback {(e: KeyboardEvent) => void} The callback that gets executed
+ * when the Escape key is pressed. The KeyboardEvent generated by the Escape
+ * keypress is passed as the only argument.
+ *
+ * @param callbackDependencies {React.DependencyList} The dependencies of the given
+ * `onEscape` callback for memoization. See `React.useCallback` for more info.
+ */
+export const useOnEscapePress = (
+  onEscape: (e: KeyboardEvent) => void,
+  callbackDependencies?: React.DependencyList
+): void => {
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    console.log('called')
+    if (e.key === 'Escape' && !e.defaultPrevented) {
+      for (let i = handlers.length - 1; i >= 0; --i) {
+        handlers[i](e)
+        if (e.defaultPrevented) {
+          break
+        }
       }
-    },
-    [onEscape]
-  )
+    }
+  }, [])
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const escapeCallback = useCallback(onEscape, callbackDependencies ?? [onEscape])
```

Comment:
`callbackDependencies ?? [onEscape]` is an interesting change from what we have been doing elsewhere.  I think the more general pattern has been `[onEscape, ...(callbackDependencies ?? []]`.  Should we be switching more to this style where you can completely override the dependencies?